varats/varats/tools/driver_casestudy.py

```diff
@@ -579,13 +610,42 @@ def remove_error_result_files() -> None:
                     #  deleted files.
 
     def remove_result_files_by_regex() -> None:
-        pass
-        # TODO: Implement the removal of files specified by the user
-        #  https://github.com/se-passau/VaRA/issues/488
+        filter = args["filter"]
+        if filter == "":
+            parser.error("Specify a regex filter with --filter or -f")
+        result_dir_paths = find_result_dir_paths_of_projects()
+
+        for result_dir_path in result_dir_paths:
+            result_file_names = os.listdir(result_dir_path)
+            todelete = []
```

Comment:
```suggestion
            files_to_delete: tp.List[?] = []
```
Maybe `files_to_delete` is a clearer name.
+ For empty list constructions we should add a type annotation to specify the type, so miss use later would be caught.